### ISSUE TITLE
`CA1010: Collections should implement generic interface` false positive when inheriting from `Dictionary<TKey, TValue>`

### ISSUE THREAD

**Issue by julealgon** [dotnet/roslyn-analyzers#7780]
I'm starting to use the `Microsoft.Extensions.StaticAnalysis` NuGet, which enables `CA1010`. However, it is firing on a piece of code that I believe has no issues at all.

```csharp
public sealed class LevelRemappingLoggerOptions : Dictionary<LogLevel, LogLevel>;
```

Here I get:
> Type 'LevelRemappingLoggerOptions' directly or indirectly inherits 'IDictionary' without implementing 'IDictionary<TKey, TValue>'. Publicly-visible types should implement the generic version to broaden usability.

This makes no sense to me, because it already inherits from the generic `Dictionary<TKey, TValue>` class, which in turn already implements `IDictionary<TKey, TValue>`. I don't see how it should be firing here.

I noticed that if I switch to inheriting from a different concrete collection, such as `List<T>`, the problem doesn't manifest: for whatever reason, this seems specific to `Dictionary<TKey, TValue>`.

For now, I'll just suppress the rule in this particular case, but it would be nice if it was fixed.
